src/Inkluzitron/Modules/Badges/BadgesModule.cs

```diff
@@ -0,0 +1,172 @@
+﻿using Discord;
+using Discord.Commands;
+using ImageMagick;
+using Inkluzitron.Data;
+using Inkluzitron.Data.Entities;
+using Inkluzitron.Extensions;
+using Inkluzitron.Models.Settings;
+using Inkluzitron.Services;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Inkluzitron.Modules.Badges
+{
+    [RequireUserPermission(GuildPermission.ManageRoles)]
+    [Name("Správa odznaků")]
+    public class BadgesModule : ModuleBase
+    {
+        private ReactionSettings ReactionSettings { get; }
+        private IHttpClientFactory HttpClientFactory { get; }
+        private DatabaseFactory DatabaseFactory { get; }
+        private BotDatabaseContext DbContext { get; set; }
+
+        public BadgesModule(DatabaseFactory databaseFactory, IHttpClientFactory httpClientFactory,
+            ReactionSettings reactionSettings)
+        {
+            DatabaseFactory = databaseFactory;
+            HttpClientFactory = httpClientFactory;
+            ReactionSettings = reactionSettings;
+        }
+
+        protected override void BeforeExecute(CommandInfo command)
+        {
+            DbContext = DatabaseFactory.Create();
+            base.BeforeExecute(command);
+        }
+
+        protected override void AfterExecute(CommandInfo command)
+        {
+            DbContext?.Dispose();
+            base.AfterExecute(command);
+        }
+
+        [Command("badge list")]
+        [Alias("badges")]
+        [Summary("Vypíše všechny odznaky a uživatele, kteří odznak získali.")]
+        public async Task ListBadgesAsync()
+        {
+            var msg = new StringBuilder();
+
+            var badges = DbContext.Badges.AsQueryable().Include(b => b.Users);
+            foreach (var badge in badges)
+            {
+                msg.AppendLine($"\n**{badge.Name} ({badge.Id})**");
+
+                if (badge.Description != null)
+                {
+                    msg.AppendLine($"*{badge.Description}*");
+                }
+
+                if (badge.Users.Count > 0)
+                {
+                    msg.AppendLine(string.Join(", ", badge.Users.Select(u => u.Name)));
```

Comment:
Nejsou ošetřené divné jména uživatelů (s odkazy) atd. Nebylo by špatné ošetřit i názvy/ID odznaků.

Reply:
Tru, na to jsem zapomněl